core/lib/spree/core/importer/order.rb

```diff
@@ -22,17 +22,20 @@ def self.import(user, params)
             create_adjustments_from_params(params.delete(:adjustments_attributes), order)
             create_payments_from_params(params.delete(:payments_attributes), order)
 
-            if completed_at = params.delete(:completed_at)
-              order.completed_at = completed_at
-              order.state = 'complete'
-            end
-
             params.delete(:user_id) unless user.try(:has_spree_role?, "admin") && params.key?(:user_id)
 
+            completed_at = params.delete(:completed_at)
+
             order.update_attributes!(params)
 
             order.create_proposed_shipments unless shipments_attrs.present?
 
+            if completed_at
+              order.completed_at = completed_at
+              order.state = 'complete'
+              order.save!
+            end
```

Comment:
it actually is necessary because create_proposed_shipments (just above) cannot run for a completed order, so if the order importer truly wants to set the order as completed, it has to happen after that check. there were failing specs around this expected behavior